### PR TITLE
Add SubParams.TrialFromPlan and SubItemsParams.ClearUsage

### PR DESCRIPTION
**Pull request by mholt**
This was simpler than I thought it'd be (and it seems to work!) so here is my pull request to implement #545. I don't have a whole lot of time, but let me know if there's anything I missed!

### CODE REVIEW

File: sub.go [hunk 1]
```diff
@@ -69,6 +70,7 @@ func (p *SubParams) AppendTo(body *form.Values, keyParts []string) {
 // For more details see https://stripe.com/docs/api#create_subscription and https://stripe.com/docs/api#update_subscription.
 type SubItemsParams struct {
 	Params       `form:"*"`
+	ClearUsage   bool   `form:"clear_usage"`
```
[the remark this answers is not in the document]
Comment (mholt) [reply sub.go]: It's here, for updating a subscription: https://stripe.com/docs/api#update_subscription-items-clear_usage -- correct me if I have this in the wrong place. But I just tested it and it worked today what didn't yesterday 😶 

Thanks for the quick look, by the way!

[the remark this answers is not in the document]
Comment (remi-stripe) [reply sub.go]: Ah gotcha, I totally missed we added that property!

**Comment by remi-stripe** [comment]
This looks good to me but I'll assign to someone for review. Thanks a lot for the quick PR!

r? @brandur-stripe 
cc @stripe/api-libraries

File: sub.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (mholt) [reply sub.go]: No problem -- you guys have a huge API and frankly, it's a real joy to work with. Thanks for setting the example for API providers everywhere!

**Comment by brandur-stripe** [comment]
LGTM. Thanks @mholt and @remi-stripe!

(And sorry Remi, yet another rebase on your mega-patch.)

**Comment by brandur-stripe** [comment]
Released as 30.6.0.

**Comment by mholt** [comment]
Thanks guys! My experience with Stripe has been phenomenal from the beginning. Your teams do a great job.

**Comment by brandur-stripe** [comment]
@mholt That's great to hear! Thanks again for the contribution here too — we love it when users send PRs ;)

**Pull request merged**